test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -602,8 +603,8 @@ class NoOpHttpFilter : public grpc_core::XdsHttpFilterImpl {
 // clock API. It's unclear if they are using the same syscall, but we do know
 // GPR round the number at millisecond-level. This creates a 1ms difference,
 // which could cause flake.
-grpc_millis NowFromCycleCounter() {
-  return grpc_timespec_to_millis_round_down(gpr_now(GPR_CLOCK_MONOTONIC));
+grpc_core::Timestamp NowFromCycleCounter() {
+  return grpc_core::Timestamp(gpr_now(GPR_CLOCK_MONOTONIC));
```

Comment:
This was previously rounding down, but the `Timestamp` ctor rounds up.  We previously had to change this to round down to fix flakiness, as per #27467.  Is this change going to reintroduce that flakiness?

Reply:
#27467 includes no discussion of why the rounding needed to be changed, only that it fixed things.
Is there a line of reasoning somewhere on why the rounding should be that way?

My worry here is that we're relying on some coincidence - insisting on millisecond precision in tests seems bound to produce some level of flakiness in any CI.

Reply:
I posted the investigation details in b/185138304.

> The ExecCtx::Now is rounding the timestamp down at millisecond-level, while the test is rounding the timestamp up.

The code change in this PR includes updating `ExecCtx::Now`, both of them are rounding at the same direction, that should be fine.

It was flaky, because rounding in 2 different direction led to +-1 millisecond gap between observed value and expected value.